lib/classes/Service.js

```diff
@@ -120,206 +118,174 @@ class Service {
         if (serverlessYml.defaults && serverlessYml.defaults.variableSyntax) {
           this.defaults.variableSyntax = serverlessYml.defaults.variableSyntax;
         }
-      })
-      .then(() => that.serverless.yamlParser
-        .parse(serverlessEnvYmlPath))
-      .then((serverlessEnvYmlParam) => {
-        const serverlessEnvYml = serverlessEnvYmlParam;
-
-        // safely load serverless.env.yml while avoiding
-        // reference errors
-        serverlessEnvYml.vars = serverlessEnvYml.vars || {};
-        serverlessEnvYml.stages = serverlessEnvYml.stages || {};
-        Object.keys(serverlessEnvYml.stages).forEach(stage => {
-          serverlessEnvYml.stages[stage] = serverlessEnvYml.stages[stage] || {};
-          serverlessEnvYml.stages[stage].vars = serverlessEnvYml.stages[stage].vars || {};
-          serverlessEnvYml.stages[stage].regions = serverlessEnvYml.stages[stage].regions || {};
-          Object.keys(serverlessEnvYml.stages[stage].regions).forEach(region => {
-            serverlessEnvYml.stages[stage].regions[region] =
-              serverlessEnvYml.stages[stage].regions[region] || {};
-            serverlessEnvYml.stages[stage].regions[region].vars =
-              serverlessEnvYml.stages[stage].regions[region].vars || {};
-          });
-        });
 
-        that.environment = serverlessEnvYml;
+        if (serverlessYml.defaults) {
+          const warningMessage = [
+            'Deprecation Notice: the "defaults" property in serverless.yml',
+            ' is deprecated. The "stage", "region" & "variableSyntax" properties',
+            ' has been moved to the "provider" property instead. Please update',
+            ' your serverless.yml file asap.',
+          ].join('');
+          this.serverless.cli.log(warningMessage);
 
-        return BbPromise.resolve(that);
-      })
-      .then(() => {
-        if (!options.stage) {
-          options.stage = this.defaults.stage;
+          if (serverlessYml.defaults.stage) {
```

Comment:
Sorry for all the remarks. But first we trigger a deprecation warning and then we keep on using the `serverlessYml.defaults`?